### ISSUE TITLE
add support for Qwen 2

### ISSUE THREAD

**Issue by aniketmaurya** [Lightning-AI/litgpt#1498]
[Qwen 2](https://qwenlm.github.io/blog/qwen2/) 500M seem to generate really good result given its small size with context length of 32K.

**Comment by rasbt** [comment]
Yes, Qwen models are really really good. They were among the best in the LLM Efficiency challenge last year. We never had time to add it, but I am in favor of it. And Qwen 2 rivals Llama 3 at the moment, so definitely a good model to have.

**Comment by 02shanks** [comment]
@aniketmaurya I would like to contribute to this model. Could you please guide me through the steps? 
Thank you!

**Comment by staticpunch** [comment]
@02shanks did you already have a personal branch that added Qwen2 support?

**Comment by 02shanks** [comment]
@staticpunch not right now. Now I will start with the adding the support for Qwen 2 model and open a PR asap!

**Comment by dnhkng** [comment]
@02shanks Qwen2.5 too please!
